File: cpp/barang.cpp
#include <iostream>
#include <string>

using namespace std;

class barang {
    private:
        int id;
        string nama;
        string harga;
        string merk;
        string kualitas;

    public:
        //konstuktor kosong
        barang()
        {
            id = -1; //default id invalid
        }

        barang(int id, string nama, string harga, string merk, string kualitas) {
            this->id = id;
            this->nama = nama;
            this->harga = harga;
            this->merk = merk;
            this->kualitas = kualitas;
        }

        void setId(int id)
        {
            this->id = id;
        }

        int getId()
        {
            return this->id;
        }

        void setNama(string nama)
        {
            this->nama = nama;
        }

        string getNama()
        {
            return this->nama;
        }

        void setHarga(string harga)
        {
            this->harga = harga;
        }

        string getHarga()
        {
            return this->harga;
        }

        void setMerk(string merk)
        {
            this->merk = merk;
        }

        string getMerk()
        {
            return this->merk;
        }

        void setkualitas(string kualitas)
        {
            this->kualitas = kualitas;
        }

        string getkualitas()
        {
            return this->kualitas;
        }

        

        // Fungsi tampilkan
        void tampilkan() {
            cout << "ID       : " << id << endl;
            cout << "Nama     : " << nama << endl;
            cout << "Harga    : " << harga << endl;
            cout << "Merk     : " << merk << endl;
            cout << "Kualitas : " << kualitas << endl;
            cout << "--------------------------" << endl;
        }

        ~barang()
        {

        }

};

File: java/Barang.java
public class Barang {
    private int id;
    private String nama;
    private String harga;
    private String merk;
    private String kualitas;

    // Konstruktor kosong
    public Barang() {
        this.id = -1; // default invalid
    }

    // Konstruktor dengan parameter
    public Barang(int id, String nama, String harga, String merk, String kualitas) {
        this.id = id;
        this.nama = nama;
        this.harga = harga;
        this.merk = merk;
        this.kualitas = kualitas;
    }

    // Setter & Getter
    public void setId(int id) { this.id = id; }
    public int getId() { return id; }

    public void setNama(String nama) { this.nama = nama; }
    public String getNama() { return nama; }

    public void setHarga(String harga) { this.harga = harga; }
    public String getHarga() { return harga; }

    public void setMerk(String merk) { this.merk = merk; }
    public String getMerk() { return merk; }

    public void setKualitas(String kualitas) { this.kualitas = kualitas; }
    public String getKualitas() { return kualitas; }

    // Fungsi tampilkan
    public void tampilkan() {
        System.out.println("ID       : " + id);
        System.out.println("Nama     : " + nama);
        System.out.println("Harga    : " + harga);
        System.out.println("Merk     : " + merk);
        System.out.println("Kualitas : " + kualitas);
        System.out.println("--------------------------");
    }
}
